api/pkg/apis/v1alpha1/providers/states/k8s/k8s.go

```diff
@@ -253,7 +253,7 @@ func (s *K8sStateProvider) Upsert(ctx context.Context, entry states.UpsertReques
 			item.SetAnnotations(metadata.Annotations)
 		}
 		getResourceVersion := false
-		if v, ok := dict["spec"]; ok {
+		if v, ok := dict["spec"]; ok && !entry.Options.UpdateStateOnly {
```

Comment:
Can you please hint me why we need the new option - UpdateStateOnly?

Reply:
UpsertOption has a field UpdateStateOnly but k8s state provider didn't respect it previously. We should only update status only if UpdateStateOnly is true. You can refer to memoryStateProvider for more information.